benefits/enrollment/views.py

```diff
@@ -26,80 +26,66 @@
 logger = logging.getLogger(__name__)
 
 
+def _get_enrollment_class(agency: models.TransitAgency):
+    if agency.littlepay_config:
+        return LittlepayEnrollment
+    elif agency.switchio_config:
+        return SwitchioEnrollment
+    else:
+        raise ValueError("Transit agency does not have a Littlepay or Switchio config")
+
+
+@decorator_from_middleware(AgencySessionRequired)
 @decorator_from_middleware(EligibleSessionRequired)
 def token(request):
     """View handler for the enrollment auth token."""
-    if not session.enrollment_token_valid(request):
-        response = request_card_tokenization_access(request)
-
-        if response.status is Status.SUCCESS:
-            session.update(request, enrollment_token=response.access_token, enrollment_token_exp=response.expires_at)
-        elif response.status is Status.SYSTEM_ERROR or response.status is Status.EXCEPTION:
-            logger.debug("Error occurred while requesting access token", exc_info=response.exception)
-            sentry_sdk.capture_exception(response.exception)
-            analytics.failed_access_token_request(request, response.status_code)
-
-            if response.status is Status.SYSTEM_ERROR:
-                redirect = reverse(routes.ENROLLMENT_SYSTEM_ERROR)
-            else:
-                redirect = reverse(routes.SERVER_ERROR)
+    agency = session.agency(request)
+    enrollment = _get_enrollment_class(agency)()
 
-            data = {"redirect": redirect}
-            return JsonResponse(data)
+    response, data = enrollment.initiate_tokenization(request)
 
-    data = {"token": session.enrollment_token(request)}
+    if response.status is Status.SYSTEM_ERROR or response.status is Status.EXCEPTION:
+        logger.debug("Error occurred while initiating tokenization", exc_info=response.exception)
+        sentry_sdk.capture_exception(response.exception)
+        analytics.failed_access_token_request(request, response.status_code)
+
+        if response.status is Status.SYSTEM_ERROR:
+            redirect = reverse(routes.ENROLLMENT_SYSTEM_ERROR)
+        else:
+            redirect = reverse(routes.SERVER_ERROR)
+
+        data = {"redirect": redirect}
 
     return JsonResponse(data)
 
 
+@decorator_from_middleware(AgencySessionRequired)
 @decorator_from_middleware(EligibleSessionRequired)
 def index(request):
     """View handler for the enrollment landing page."""
     session.update(request, origin=reverse(routes.ENROLLMENT_INDEX))
 
+    agency = session.agency(request)
+    enrollment = _get_enrollment_class(agency)()
+
+    # this is what we have to do until we introduce separate Django apps that have littlepay and switchio views
+    if isinstance(enrollment, LittlepayEnrollment):
+        return _index_for_littlepay(request, enrollment)
+    else:  # is instance of SwitchioEnrollment
+        return _index_for_switchio(request, enrollment)
```

Comment:
At first, I tried to have a method on `LittlepayEnrollment`/`SwitchioEnrollment` that would return the right view function to use, but it was circular-import galore, so I resorted to this.

I think if we had `littlepay` and `switchio` apps, that might give us a cleaner way to invoke the appropriate view logic.

Reply:
I think we should not introduce technical debt like this when you have a better idea/approach in mind already.